lib/pengine/utils.c

```diff
@@ -804,21 +804,29 @@ unpack_operation(action_t * action, xmlNode * xml_obj, resource_t * container,
         action->on_fail = action_fail_restart_container;
         value = "restart container (and possibly migrate) (default)";
 
-    /* for baremetal remote nodes, ensure that a recurring monitor operation failure
-     * defaults to either fencing the remote-node for recovery, or at least
-     * attempting to recover the the connection when fencing is disabled. */
+    /* for barmetal remote nodes, ensure that any failure that results in
+     * dropping an active connection to a remote node results in fencing of
+     * the remote node.
+     *
+     * There are only two action failures that don't result in fencing.
+     * 1. probes - probe failures are expected.
```

Comment:
okay, so for cluster nodes we have 'startup-fencing=true|false". it makes sense to default this option to 'true' for cluster nodes because we can't do anything about an unseen cluster node except fence...

for example, we can't reach out to it and say "hey cluster node are you there?" but with remote nodes things are different. remote nodes don't joint themselves into the cluster, the cluster nodes have to opt them in.  It seems odd to immediately fence a new remote node just because the cluster learned about it when the cluster is also responsible for reaching out and integrating it.  I'd rather see us attempt to connect to an unseen remote node first. If an unseen remote node can not be connected to, then we should fence. This might already be happening. I need to test this to know for sure.

After that initial connection, there shouldn't be any amount of "cleanup" that will prevent us from knowing if a connection disappeared or not. If there is, then that is a bug. Cleanup never results in actually dropping the connection, it just involves the crmd's history hash table to get cleared out. During the time between the history being cleared and a failure in the connection, the recurring monitors are still going and a monitor failure will still be reported if the connection is lost. From there the remote node should get fenced indicating that it safely stopped.